### PR TITLE
Tas 1565/saved jobs page

### PR DESCRIPTION
**Pull request by marjan-ara**

### CODE REVIEW

File: src/Nowruz/modules/Jobs/modules/JobListing/useJobListing.tsx [hunk 1]
```diff
@@ -1,55 +1,56 @@
...
+      await Promise.all([fetchMore(), getSkills()]);
+      setLoading(false);
+    } catch (error) {
+      setLoading(false);
+    }
+  };
+
+  const fetchMore = async () => {
+    if (!isMount) {
+      const data = await jobs({ page: page, status: 'ACTIVE', limit: PER_PAGE });
```
Comment (satanmourner) [inline src/Nowruz/modules/Jobs/modules/JobListing/useJobListing.tsx:38]: `try/catch` 
FMI: don't we have global error handling?
I didn't see any and if it's so, please let me know to create an issue to handle globally with `Material Alert` for these catching globally.
it's easier for us to skip these try/catch clauses and not use them every time unless we need another handler for errors so we can overwrite by using these clauses.

File: src/Nowruz/modules/Jobs/modules/savedJobListing/useSavedJobListing.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      await Promise.all([fetchMore(), getSkills()]);
+      setLoading(false);
+    } catch (error) {
+      setLoading(false);
+    }
+  };
+
+  const fetchMore = async () => {
+    if (!isMount) {
+      const data = await markedJobs({ page: page, 'filter.marked_as': 'SAVE', limit: 5 });
```
Comment (satanmourner) [inline src/Nowruz/modules/Jobs/modules/savedJobListing/useSavedJobListing.tsx:38]: `try/catch`

**Review by satanmourner — CHANGES_REQUESTED** [review]

File: src/Nowruz/modules/Jobs/components/JobListingCard/useJobListingCard.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      }
+    } catch (e) {
+      console.log('error in bookmark click', e);
+    }
+  };
+
+  const handleNotInterested = async () => {
+    try {
+      if (jobVal.saved) {
+        const res = await removeMark(job.id);
```
Comment (satanmourner) [inline src/Nowruz/modules/Jobs/components/JobListingCard/useJobListingCard.tsx:56]: why do you need constant `res` if you don't use it later, you can use `await removeMark(job.id)` directly without pouring into a variable?
line-58 too

**Review by satanmourner — CHANGES_REQUESTED** [review]

File: src/Nowruz/modules/Jobs/components/JobListingCard/index.tsx [hunk 4]
```diff
@@ -75,8 +66,35 @@ export const JobListingCard: React.FC<JobListingCardProps> = ({ job }) => {
...
+              e.stopPropagation();
+              handleNotInterested();
+            }}
+          />
+        )}
+        {displaySave && (
+          <IconButton
+            iconName={jobVal.saved ? '' : 'bookmark'}
+            disabled={jobVal.not_interested}
+            img={jobVal.saved ? <img src="/icons/nowruz/green-bookmark.svg" alt="" /> : ''}
```
[the remark this answers is not in the document]
Comment (marjan-ara) [reply src/Nowruz/modules/Jobs/components/JobListingCard/index.tsx:86]: We don't have them in icon pack already. Let's discuss it in the next sprint

File: src/Nowruz/modules/Jobs/modules/JobListing/useJobListing.tsx (hunk 1 above)
Comment (marjan-ara) [reply src/Nowruz/modules/Jobs/modules/JobListing/useJobListing.tsx:38]: Yes, we had it in V2, but in Nowruz it is not implemented yet. We should do it

**Review by satanmourner — APPROVED** [review]

**Pull request merged**